### PR TITLE
Add specs

### PR DESCRIPTION
**Pull request by mattkorb**

### CODE REVIEW

File: inst/validation/specs.R [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Use a list to declare the specs
+# nolint start
```
Comment (iglauss) [inline inst/validation/specs.R:2]: Should we restrict to line_length_linter?

File: inst/validation/specs.R [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+specs_list <- list
+
+listing <- specs_list(
+  "display_listing" = "dv.listings displays a dataset in a tabular form",
+  "listing_selection" = "dv.listings includes a dropdown menu to select which listing to be shown.",
+  "listings_label" = "dv.listings displays the label of a listing if available. The label is concatenated to the listing’s dataset name and the resulting strings are provided as choices in the listings dropdown menu.",
```
Comment (iglauss) [inline inst/validation/specs.R:8]:
```suggestion
  "listings_label" = "dv.listings displays the label of a dataset if available. The label is concatenated to the dataset name and the resulting strings are provided as choices in the module's dropdown menu.",
```

File: inst/validation/specs.R [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Use a list to declare the specs
+# nolint start
+specs_list <- list
+
+listing <- specs_list(
+  "display_listing" = "dv.listings displays a dataset in a tabular form",
```
Comment (iglauss) [inline inst/validation/specs.R:6]:
```suggestion
  "display_listing" = "dv.listings displays a dataset as listing",
```

File: inst/validation/specs.R [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "listing_selection" = "dv.listings includes a dropdown menu to select which listing to be shown.",
```
Comment (iglauss) [inline inst/validation/specs.R:7]:
```suggestion
  "listing_selection" = "dv.listings includes a dropdown menu to select which dataset to be shown.",
```

File: inst/validation/specs.R [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  "listings_label" = "dv.listings displays the label of a listing if available. The label is concatenated to the listing’s dataset name and the resulting strings are provided as choices in the listings dropdown menu.",
+  "column_selection" = "dv.listings includes a dropdown menu to select the columns from the selected listing to be shown and arrange their order.",
+  "column_label" = "dv.listings displays the column labels of a listing if available. Column names are pasted together with their label. These extended column titles replace the original column names, so that they are visible in the listings display and column dropdown menu.",
```
Comment (iglauss) [inline inst/validation/specs.R:10]:
```suggestion
  "column_label" = "dv.listings displays extended column headers consisting of the variable name pasted together with its label, if available. These extended column headers replace the original variable names in the column dropdown menu.",
```

File: inst/validation/specs.R [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  "sorting_columns" = "dv.listings includes sorting functionality for each of the different variables included in the dataset",
```
Comment (iglauss) [inline inst/validation/specs.R:11]:
```suggestion
  "sorting_columns" = "dv.listings includes sorting functionality for each of the column.",
```

File: inst/validation/specs.R [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  "restore_row_order" = "dv.listings includes a button to restore the row order of a listing to the state as it is in the original data.",
+  "default_vars" = "If pre-specifications for default columns are available, dv.listings will display them at app launch for the respective listing. If not, dv.listings will show the first six columns of the listing - or all columns, in case the number of columns is less than six.",
+  "retain_last_selection" = "dv.listings can remember and retain the last column selections after switching listings during the current session. It also restores the ",
```
Comment (iglauss) [inline inst/validation/specs.R:14]: it also restores the...?

**Review by iglauss — CHANGES_REQUESTED** [review]

File: inst/validation/specs.R (hunk 1 above)
Comment (mattkorb) [reply inst/validation/specs.R:2]: Yes, we can also do it like this :)

File: inst/validation/specs.R (hunk 7 above)
Comment (mattkorb) [reply inst/validation/specs.R:14]: Oh must have missed it. Will complete the sentence. :)

**Review by iglauss — APPROVED** [review]

**Pull request merged**